src/components/VideoThumbnail/VideoThumbnail.tsx

```diff
@@ -56,8 +66,30 @@ export function VideoThumbnail({
     );
   }
 
+  const progressValue = calculateProgress(videoLength, videoProgress);
+
   const timeStampMarkup = videoLength ? (
-    <p className={styles.Timestamp}>{secondsToTimestamp(videoLength)}</p>
+    <p
+      className={classNames(
+        styles.Timestamp,
+        progressValue && styles.WithProgress,
+      )}
+    >
+      {secondsToTimestamp(videoLength)}
+    </p>
+  ) : null;
+
+  const progressMarkup = progressValue ? (
+    <div className={styles.Progress}>
+      <progress
+        className={styles.ProgressBar}
+        value={progressValue}
+        max="100"
+      />
+      <div className={styles.Indicator} style={{width: `${progressValue}%`}}>
```

Comment:
If this will ever be changing live (as in: the indicator moves as  while the user plays the video), use `transform: scaleX(<value between 0 and 1>)` instead of `width`, as `transform` animates more efficiently than `width`.

Also set these styles on the indicator:

```scss
  width: 100%; // (rather than 0)
  transform-origin: left;
  transform: scaleX(0);
```

Reply:
Great suggestion! Did not have use-case for live update yet, but definitely better to have it! 👍